packages/mds-db/index.ts

```diff
@@ -89,6 +92,134 @@ async function initialize() {
+      first = first[0]
+    }
+
+    first = first[0]
+    switch (typeof first) {
+      case "object":
+        type = "JSON" + type
+        break
+      case "string":
+        type = "varchar(31)" + type
+        break
+      default:
+        type = typeof first + type
+    }
+
+    let commaized_content = commaize(
+      data.map(data_element => stringify(data_element, `'`, true)),
+      ``
+    )
+    let cast = !nested && type !== "[]"
+    return `${cast ? "CAST(" : ""}${
+      nested ? "" : "ARRAY"
+    }[${commaized_content}]${cast ? ` AS ${type})` : ""}`
+  } else if (typeof data === "object") {
+    return `${quote}${JSON.stringify(data)}${quote}`
+  } else {
+    return `${quote}${data}${quote}`
+  }
+}
+
+async function runQuery(query: any) {
+  const client = await getWriteableClient()
+  let results = await client.query(query)
+  return results.rows
+}
+
+async function getStates(
+  provider_id: any,
+  start_time: any = 0,
+  end_time: any = Date.now()
+) {
+  let query = `SELECT * FROM reports_device_states WHERE utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  //let query = `SELECT * FROM reports_device_states WHERE provider_id = ${provider_id} AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  return runQuery(query)
+}
+
+async function getTripCount(
+  provider_id: any,
+  start_time: any = 0,
+  end_time: any = Date.now()
+) {
+  let query = `SELECT count(DISTINCT trip_id) FROM reports_device_states WHERE type = 'event' AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  //let query = `SELECT count(DISTINCT trip_id) FROM reports_device_states WHERE provider_id = ${provider_id} AND type = 'event' AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  return runQuery(query)
+}
+
+async function getVehicleTripCount(
+  device_id: any,
+  start_time: any = 0,
+  end_time: any = Date.now()
+) {
+  let query = `SELECT count(DISTINCT trip_id) FROM reports_device_states WHERE type = 'event' AND device_id = '${device_id}' AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  return runQuery(query)
+}
+
+async function getLateEventCount(
+  provider_id: any,
+  events: any,
+  start_time: any = 0,
+  end_time: any = Date.now()
+) {
+  let query = `SELECT count(*) FROM reports_device_states WHERE event_type IN ${events} AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  //let query = `SELECT count(*) FROM reports_device_states WHERE provider_id = ${provider_id} AND event_type IN ${events} AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  return runQuery(query)
+}
+
+async function getTrips(
+  provider_id: any,
+  start_time: any = 0,
+  end_time: any = Date.now()
+) {
+  let query = `SELECT * FROM reports_trips WHERE end_time BETWEEN ${start_time} AND ${end_time}`
+  //let query = `SELECT * FROM reports_trips WHERE provider_id = ${provider_id} AND utc_epoch BETWEEN ${start_time} AND ${end_time}`
+  return runQuery(query)
+}
+
+async function insert(table_name: TABLE_NAME, data: { [x: string]: any }) {
```

Comment:
The reason we have functions like `writePolicy` that then take an object of type `Policy` is that it offers certain guarantees that the object you're sticking in actually satisfies the requirements a `Policy` is supposed to have.